components/service/glean/src/test/java/mozilla/components/service/glean/StringListMetricTypeTest.kt

```diff
@@ -0,0 +1,193 @@
+                .edit()
+                .clear()
+                .apply()
+        StringListsStorageEngine.clearAllStores()
+    }
+
+    @Test
+    fun `The API must define the expected "default" storage`() {
+        // Define a 'stringMetric' string metric, which will be stored in "store1"
+        val stringListMetric = StringListMetricType(
+            disabled = false,
+            category = "telemetry",
+            lifetime = Lifetime.Ping,
+            name = "string_list_metric",
+            sendInPings = listOf("store1")
+        )
+        assertEquals(listOf("metrics"), stringListMetric.defaultStorageDestinations)
+    }
+
+    @Test
+    fun `The API saves to its storage engine`() {
+        // Define a 'stringMetric' string metric, which will be stored in "store1"
+        val stringListMetric = StringListMetricType(
+            disabled = false,
+            category = "telemetry",
+            lifetime = Lifetime.Application,
+            name = "string_list_metric",
+            sendInPings = listOf("store1")
+        )
+
+        // Record two lists using add and set
+        stringListMetric.add("value1")
+        stringListMetric.add("value2")
+        stringListMetric.add("value3")
+
+        // Check that data was properly recorded.
+        val snapshot = StringListsStorageEngine.getSnapshot(storeName = "store1", clearStore = false)
+        assertEquals(1, snapshot!!.size)
+        assertEquals(true, snapshot.containsKey("telemetry.string_list_metric"))
+        assertEquals("value1", snapshot.get("telemetry.string_list_metric")?.get(0))
+        assertEquals("value2", snapshot.get("telemetry.string_list_metric")?.get(1))
+        assertEquals("value3", snapshot.get("telemetry.string_list_metric")?.get(2))
+
+        stringListMetric.set(listOf("other1", "other2", "other3"))
+        // Check that data was properly recorded.
+        val snapshot2 = StringListsStorageEngine.getSnapshot(storeName = "store1", clearStore = false)
+        assertEquals(1, snapshot2!!.size)
+        assertEquals(true, snapshot2.containsKey("telemetry.string_list_metric"))
+        assertEquals("other1", snapshot2.get("telemetry.string_list_metric")?.get(0))
+        assertEquals("other2", snapshot2.get("telemetry.string_list_metric")?.get(1))
+        assertEquals("other3", snapshot2.get("telemetry.string_list_metric")?.get(2))
+    }
+
+    @Test
+    fun `The API truncates long string values`() {
+        // Define a 'stringMetric' string metric, which will be stored in "store1"
+        val stringListMetric = StringListMetricType(
+            disabled = false,
+            category = "telemetry",
+            lifetime = Lifetime.Application,
+            name = "string_metric",
+            sendInPings = listOf("store1")
+        )
+
+        stringListMetric.add("0123456789012345678901234567890123456789012345678901234567890123456789")
```

Comment:
This (and other instances) could become something like 

```
val bigString = "a".repeat(100)
stringListMetric.add(bigString)
```

So that below, when checking for the value, you could do something like

```
assertEquals(bigString.take(StringListMetricType.MAX_LENGTH_VALUE, snapshot....)
```

Reply:
I like it!  I'll take care of that on my next commit.